python/tests/unit/test_healthchecks.py

```diff
@@ -0,0 +1,517 @@
+                                "fruit-one": {"append": False, "value": "banana"},
+                                "fruit-two": {"append": True, "value": "orange"},
+                                "fruit-three": {"value": "peach"},
+                            },
+                        }
+                    }
+                ]
+            ),
+            "expected": [
+                {
+                    "http_health_check": {
+                        "path": "/health",
+                        "request_headers_to_add": [
+                            {"header": {"key": "fruit-one", "value": "banana"}, "append": False},
+                            {"header": {"key": "fruit-two", "value": "orange"}, "append": True},
+                            {"header": {"key": "fruit-three", "value": "peach"}, "append": True},
+                        ],
+                    }
+                },
+            ],
+        },
+        {  # check remove headers
+            "name": "healthcheck_http_remove_headers",
+            "input": baseYaml.format(
+                [
+                    {
+                        "http_health_check": {
+                            "path": "/health",
+                            "remove_request_headers": ["fruit-one", "fruit-two", "fruit-three"],
+                        }
+                    }
+                ]
+            ),
+            "expected": [
+                {
+                    "http_health_check": {
+                        "path": "/health",
+                        "request_headers_to_remove": ["fruit-one", "fruit-two", "fruit-three"],
+                    }
+                },
+            ],
+        },
+        {  # check that we can keep a valid healthcheck while ignoring a bad one
+            "name": "healthcheck_ignore_invalid",
+            "input": baseYaml.format(
+                [
+                    {
+                        "http_health_check": {"path": "/health"},
+                    },
+                    {
+                        "http_health_check": {"hostname": "dummy.example"},
+                    },
+                ]
+            ),
+            "expected": [{"http_health_check": {"path": "/health"}}],
+        },
+        {  # Test that we throw out the health check config when there is no endpoint resolver
+            "name": "healthcheck_no_endpoint",
+            "input": noEndpointYaml.format([{"http_health_check": {"path": "/health"}}]),
+            "expected": None,
+        },
+    ]
+
+    for case in testcases:
+
+        caseYaml = typing.cast(typing.Dict[str, Any], case)["input"]  # seriously mypy???
+        testName = typing.cast(typing.Dict[str, Any], case)["name"]
+        econf = _get_envoy_config(caseYaml)
+        cluster = _get_cluster_config(econf.clusters, "cluster_coolsvcname_default")
+        assert cluster != False
+
+        expectedChecks = typing.cast(typing.Dict[str, Any], case)["expected"]
```

Comment:
You should drop these `typing.cast`s in favor of just adding type information to `testcases`. (See the comment I made further up.)